lib/azure/armrest/armrest_service.rb

```diff
@@ -253,11 +254,22 @@ def rest_head(options)
 
         def raise_api_exception(err)
           begin
-            response = JSON.parse(err.http_body)
-            code     = response['error']['code']
-            message  = response['error']['message']
+            content_type_header = err.response.headers[:content_type]
+
+            response = case content_type_header.match(%r{(?<application_type>application\/\w+)})[:application_type]
+                       when "application/json"
+                         JSON.parse(err.http_body)
+                       when "application/xml"
+                         # The XML document that is returned has Error, Code, and Message
+                         # so we need to downcase the keys to stay consistent with the
+                         # parsed-json hash.
+                         Hash.from_xml(err.http_body).deep_transform_keys(&:downcase)
```

Comment:
Hash.from_xml I believe comes from active support, so we may want to explicitly required that core extension to ensure it's loaded when we get here.

Reply:
`active_support/core_ext/hash/conversions` is already required which is what providers `from_xml`: https://github.com/ManageIQ/azure-armrest/pull/406/files#diff-4c3805a9a8d1d2c792e53c71498d5c8c0354e496069340251635f99ea606d90aR2
I added `active_support/core_ext/hash/keys` for `deep_transform_keys`